### ISSUE TITLE
If user clicks "I already have a seed" accidentally, there's currently no way to go back.

### ISSUE THREAD

**Issue by BenWestgate** [BenWestgate/Bails#31]
Clicking the X during seed restore should bring them back to the create wallet menu.

If valid shares have already been input, it should remind the user their restore progress will be lost and ask them to confirm

Later, for L1/L2 it should ask them if they'd like to save their restore progress to their Persistent Storage.

This would require writing an encryption key/passphrase (transfer key or resume key) down that is used temporarily until this wallet's seed is restored in full.

Save the last session's progress in an folder named by the unique Wallet name all at once to obfuscate the threshold.

Pad to the length of 8 shares as that is the maximum data that could be loaded without restoring the max threshold of 9. 

Add random noise equal to this length so plaintext shares can not be linked with the ciphertext saved progress.

Use public key cryptography so that entered shares can deniably encrypted to the Persistent Storage without re-entering that transport key each session. Deniable encryption means a random noise length of the plaintext is added before assymmetrically encrypting and discarded after decryption.

Private key length 128-bits. Encoded in codex32 format but with a new identifier. 256-bit ECC must be used, the public key is saved to the device.

The identifier of the ms32 private resume key can match the shares it decrypts, so that shares incorrectly saved under a wallet's restore process can be discarded or re-encrypted by the correct public key on next resume.

If the user "resumes" a wallet restore and types a correct resume key, it will be as if they had entered all their previously saved shares, even from multiple sessions. Effectively "loading" the saved restore progress.

An incorrect resume key would throw an error as the result would be random data, not valid shares. 

The resume keypair is unique per Wallet Name being restored and it is destroyed along with all ciphertexts immediately after a successful wallet restoration.

This allows multiple wallets to be restored in parallel as long as the same unique name is used the next restore session.

If saved progress is detected, the name(s) of partially restored wallets will be first presented instead of asking for a wallet_name. A last saved date & time may help further disambiguate them. The normal "Create wallet" dialog displays otherwise if no partially restored wallet is selected after are you sure you don't want to continue restoring a wallet? confirmation.

After a wallet name is chosen, the choices will be resume (decrypt and automatically enter saved shares) and Continue. After each share is entered, it will ask again if they'd like to load their saved progress (which will ask for the resume key) which they can skip. If they need to quit again, progress can be encrypted by this wallet's resume restore public key without writing or entering the private key.

The resume private key can be securely stored until the user is in a safe place to restore the seed and wallet.

in order to prevent accidentally saving mismatched shares to the same wallet restore, saved progress files can go in a general folder and it is the successful resume decryption by the resume private key with a matching threshold+identifier that decides which saves will be loaded to which wallet restore.

The resume public key is saved as the unique "wallet name" it is encrypting restore progress for.

**Comment by BenWestgate** [comment]
Resolved by:
https://github.com/BenWestgate/Bails/commit/1b6bdbc26ba8c74aaa09558de24d11ec3b49be07

Will fork the "save progress" idea into a new low priority issue as it is dependent on codex32 generation reference being approved.
